include/node.tcc

```diff
@@ -383,25 +383,30 @@ void mpm::Node<Tdim, Tdof, Tnphases>::apply_absorbing_constraint() {
   const double swave_v = std::get<2>(this->absorbing_constraints_);
 
   // Phase: Integer value of division (dir / Tdim)
-  const auto phase_n = static_cast<unsigned>(dir_n / Tdim);
-
-  if (Tdim == 2) {
-    // Determine Shear Direction
-    const unsigned dir_s = (Tdim - 1) - dir_n;
-    const auto phase_s = static_cast<unsigned>(dir_s / Tdim);
-
-    // Calculate Traction Forces
-    double mass_density_s = mass_(phase_s) / volume_(phase_s);
-    double velocity_s = velocity_(dir_s, phase_s);
-    const double traction_s = -velocity_s * mass_density_s * swave_v;
-
-    double mass_density_n = mass_(phase_n) / volume_(phase_n);
-    double velocity_n = velocity_(dir_n, phase_n);
-    const double traction_n = -velocity_n * mass_density_n * pwave_v;
- 
-    absorbing_traction_(dir_s,phase_s) = traction_s;
-    absorbing_traction_(dir_n,phase_n) = traction_n;
+  const unsigned phase = 0;
+
+  // Create Traction Value  
+  double traction = 10;
+
+  //if (Tdim == 2) {
+  for (unsigned dir = 0; dir < Tdim; ++dir) {
+  	// Calculate Traction Forces
+	if (dir == dir_n) {
+    	double mass_density = mass_(phase) / volume_(phase);
```

Comment:
indent

Please define `mass_density` an d`velocity` outside the if statement.
```
double mass_density = 0;
double velocity = 0;
if (dir == dir_n) {
mass_density = ...
velocity - ...
else 
...
```

Reply:
Yeah, I realized this after commiting the changes, its been fixed